src/utils.ts

```diff
@@ -476,24 +484,32 @@ export class Utils {
     }
   }
 
-  public static initEvent<T>(e: DragEvent | MouseEvent, info: { type: string; target?: EventTarget }): T {
-    const evt = { type: info.type };
-    const obj = {
-      button: 0,
-      which: 0,
-      buttons: 1,
-      bubbles: true,
-      cancelable: true,
-      target: info.target ? info.target : e.target
-    };
-    // don't check for `instanceof DragEvent` as Safari use MouseEvent #1540
-    if ((e as DragEvent).dataTransfer) {
-      evt['dataTransfer'] = (e as DragEvent).dataTransfer; // workaround 'readonly' field.
+    public static initEvent<T extends DragEvent | MouseEvent>(e: DragEvent | MouseEvent, info: { type: string; target?: EventTarget }): GridStackDragMouseEvent<T> {
+        return {
+            type: info.type,
+            button: 0,
+            buttons: 1,
+            bubbles: true,
+            cancelable: true,
+            target: info?.target || e.target,
+            ...((e as DragEvent).dataTransfer && {
+                dataTransfer: (e as DragEvent).dataTransfer
+            } as any),
+            // keys

```

Comment:
not clear how that is better than the compact methods that were there before...

Reply:
this part might not be clearer indeed.  I can put that condition outside

The actual part that is not clear is the forEach section
imo .forEach shouldn't be used to modify a variable that is external to it's scope as there is reduce for that.
also regarding minification and interpretation, I'm not sure the use of strings is very efficient.

Reply:
.forEach is just looping, not really a function per say... we use it in many places to deal with list of attributes - more compact code wise and doesn't suffer copy/paste mismatch (left and right side). please restore, unless ...() is used to get all those (and then some).

Reply:
I found literature on why using .forEach for mutating data outside of the scope of the function you pass to it is unpure and should be avoided:

https://www.bennadel.com/blog/2992-mutating-an-array-during-foreach-iteration-in-javascript.htm

basically, using a regular for loop is safer. A functional programming approach is to use reduce.. which is more complicated, but you could create a helper function or use something like [pick](https://lodash.com/docs/4.17.15#pick) from lodash